pynest/nest/tests/test_connect_arrays_mpi.py

```diff
@@ -0,0 +1,142 @@
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess as sp
+import unittest
+import nest
+import numpy as np
+
+try:
+    from mpi4py import MPI
+    HAVE_MPI4PY = True
+except ImportError:
+    HAVE_MPI4PY = False
+
+HAVE_MPI = nest.ll_api.sli_func("statusdict/have_mpi ::")
+MULTIPLE_PROCESSES = nest.NumProcesses() > 1
+
+
+@unittest.skipIf(not HAVE_MPI, 'NEST was compiled without MPI')
+@unittest.skipIf(not HAVE_MPI4PY, 'mpi4py is not available')
+class ConnectArraysMPICase(unittest.TestCase):
+    """
+    This TestCase uses mpi4py to collect and assert results from all
+    processes, and is supposed to only be run with multiple processes.
+    If running with nosetests or pytest, this TestCase is ignored when
+    run with a single process, and called from TestConnectArraysMPI using
+    multiple processes.
+    """
+    non_unique = np.array([1, 1, 3, 5, 4, 5, 9, 7, 2, 8], dtype=np.uint64)
+    comm = MPI.COMM_WORLD
+
+    # With nosetests, only run these tests if using multiple processes
+    __test__ = MULTIPLE_PROCESSES
+
+    def assert_connections(self, expected_sources, expected_targets, rule='one_to_one'):
+        """Gather connections from all processes and assert against expected connections"""
+        conns = nest.GetConnections()
+        projections = [[s, t] for s, t in zip(conns.source, conns.target)]
+        if rule == 'one_to_one':
+            expected_projections = np.array([[s, t] for s, t in zip(expected_sources, expected_targets)])
+        elif rule == 'all_to_all':
+            expected_projections = np.array([[s, t] for s in expected_sources for t in expected_targets])
+        else:
+            self.assertFalse(True, 'rule={} is not valid'.format(rule))
+
+        recv_projections = self.comm.gather(projections, root=0)
+        if self.comm.Get_rank() == 0:
+            # Flatten the projection lists to a single list of projections
+            recv_projections = np.array([proj for part in recv_projections for proj in part])
+            # Results must be sorted to make comparison possible
+            np.testing.assert_array_equal(np.sort(recv_projections, axis=0), np.sort(expected_projections, axis=0))
+        else:
+            self.assertIsNone(recv_projections)
+
+    def setUp(self):
+        nest.ResetKernel()
+
+    def test_connect_arrays_unique(self):
+        """Connecting NumPy arrays of unique node IDs with MPI"""
+        n = 10
+        nest.Create('iaf_psc_alpha', n)
+        sources = np.arange(1, n+1, dtype=np.uint64)
+        targets = np.arange(1, n+1, dtype=np.uint64)
+        weights = 1.5
+        delays = 1.4
```

Comment:
why provide them if they are not checked?

Reply:
True, I've removed them.

Reply:
Mhh... maybe I should have phrased this differently, sorry ^^"
I meant: shouldn't there be a check to make sure that synaptic properties are handled correctly in this case? (maybe even with array properties)
Or is this unnecessary (maybe the underlying code reuses the same functions as the normal Connect)?

Reply:
Ah, right. While implementing testing of weights and delays I found a bug in `connect_arrays()` that should be fixed first. See #1781.

Reply:
Now that #1781 is merged, I've added tests of weights and delays.